x/stakeibc/keeper/lsm.go

```diff
@@ -166,6 +173,25 @@ func (k Keeper) GetValidatorFromLSMTokenDenom(denom string, validators []*types.
 		"validator (%s) is not registered in the Stride validator set", validatorAddress)
 }
 
+// Given an LSMToken representing a number of delegator shares, returns the stToken coin
+// using the validator's exchange rate and the host zone redemption rate
+//
+//	StTokens = LSMTokenShares * Validator Exchange Rate / Redemption Rate
+//
+// Note: in the event of a slash query, these tokens will be minted only if the
+// validator's exchange rate did not change
+func (k Keeper) CalculateLSMStToken(liquidStakedShares sdkmath.Int, lsmLiquidStake types.LSMLiquidStake) sdk.Coin {
+	hostZone := lsmLiquidStake.HostZone
+	validator := lsmLiquidStake.Validator
+
+	lsmTokenTokens := sdk.NewDecFromInt(liquidStakedShares).Mul(validator.InternalSharesToTokensRate)
+	stDenom := types.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
```

Comment:
- pico nit: move the denom line down below the calc ends to make the calc logic slightly easier to inspect
- pico nit: why `lsmTokenTokens` instead of `lsmTokens`?

Reply:
great suggestion on first bullet. 

`lsmTokenTokens` is kind of a typo - I used to have to variables `lsmTokensShares` and `lsmTokenTokens` but the shares one was replaced with `liquidStakedShares`. I'm going to update it to `liquidStakedTokens`